wondrous-app/components/OnboardingDao/Review/component.tsx

```diff
@@ -0,0 +1,127 @@
+import EditIcon from 'components/Icons/edit.svg';
+import { Error } from 'components/OnboardingDao/styles';
+import { FormikValues, useField, useFormikContext } from 'formik';
+import { useState } from 'react';
+import {
+  Category,
+  ChildrenWrapper,
+  EditButton,
+  EditButtonText,
+  EditInput,
+  EditInputMulti,
+  EditInputWrapper,
+  InputErrorWrapper,
+  ItemWrapper,
+  LabelText,
+  LabelWrapper,
+  Logo,
+  Text,
+  TextAndInputWrapper,
+  Wrapper,
+} from './styles';
+
+const Item = ({ label, children }) => {
+  return (
+    <ItemWrapper>
+      <LabelWrapper>
+        <LabelText>{label}</LabelText>
+      </LabelWrapper>
+      <ChildrenWrapper>{children}</ChildrenWrapper>
+    </ItemWrapper>
+  );
+};
+
+const ItemWithEdit = ({ value, field, EditComponent }) => {
+  const [isEditing, setIsEditing] = useState(false);
+  return (
+    <Item label={field.label}>
+      <TextAndInputWrapper>
+        {isEditing ? (
+          <EditComponent setIsEditing={setIsEditing} {...field} />
+        ) : (
+          <Text onClick={() => setIsEditing(true)}>{value}</Text>
+        )}
+      </TextAndInputWrapper>
+    </Item>
+  );
+};
+
+const Edit = ({ onClick }) => {
+  return (
+    <EditButton onClick={onClick}>
+      <EditIcon /> <EditButtonText>Edit</EditButtonText>
+    </EditButton>
+  );
+};
+
+const EditName = ({ setIsEditing, ...props }) => {
+  const [field, meta, helpers] = useField(props.name);
+  return (
+    <InputErrorWrapper>
+      <EditInputWrapper>
+        <EditInput {...field} {...props} value={field.value} onChange={(e) => helpers.setValue(e.target.value)} />
+        <Edit
+          onClick={() => {
+            if (!meta.error) setIsEditing(false);
+          }}
+        />
+      </EditInputWrapper>
+      {meta.touched && meta.error && <Error>{meta.error}</Error>}
+    </InputErrorWrapper>
+  );
+};
+
+const EditDescription = ({ setIsEditing, ...props }) => {
+  const [field, meta, helpers] = useField(props.name);
+  return (
+    <InputErrorWrapper>
+      <EditInputWrapper>
+        <EditInputMulti
+          {...field}
+          {...props}
+          value={field.value}
+          onChange={(e) => {
+            const value = e.target.value;
```

Comment:
let's set this as a separate function ( reason: it's more manageable, these functions can grow overtime and it's easier to maintain them when having them out of the render ) also at some point we'll want to have unit tests and it will be easier to test it